magma/utils.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+################################################
+#   Libraries
+################################################
+
+################################################
+#   Functions
+################################################
+#TODO: description -- small utility fxns and
+# object attribute checking
+
+#TODO: following 2 fxn not used elsewhere but could be in future
+def check_variable_type(variable, intended_type):
+    """
+    Checks that given variable is of the intended type.
+
+    :param variable: variable to be checked
+    :type variable: data type
+    :param intended_type: the variable type that is intended
+    :type intended_type: data type
+    :return: True if the variable if of the intended_type, else False
+    :rtype: bool
+    """
+    if not isinstance(variable, intended_type):
+        return False
+    else:
+        return True
+
+def check_list_elements_type(list_to_check, intended_type):
+    """
+    Checks that all elements in  list are of a given type.
+    Raises Exception if not all elements are strings.
+
+    :param list_to_check: list to be checked
+    :type list_to_check: list
+    :param intended_type: the variable type that is intended
+    :type intended_type: data type
+    :return: True if all elements of list_to_check are of the intended_type, else False
+    :rtype: bool
+    """
+    # check that all elements in list are strings
+    if not all(isinstance(element, intended_type) for element in list_to_check):
+        return False
+    else:
+        return True
+
+def check_presence_of_attributes(input_object, attributes_to_check=None):
+    """
+    Takes in an object and a list of attributes, checks that those attributes are present
+    in this object
+
+    :param input_object: object to check
+    :type input_object: object (instance of some class)
+    :param attributes_to_check: list of attributes to check
+    :type attributes_to_check: list[str]
+    :return: None, if there are no attributes to check
+    :return: None, if all specified attributes are present
+    :raises ValueError: if input_object doesn't have a specified attribute
+    """
+    #TODO: make the next three commands its own helper function? I repeat variations
+    # several times
+    if attributes_to_check is None:
+        return
+
+    for attribute in attributes_to_check:
+        try:
+            getattr(input_object, attribute)
+        except AttributeError as e:
+            raise ValueError("Object validation error, {0}\n"
+                                .format(e.args[0]))
+
+def set_list_attributes(input_object, attributes_to_set=None):
```

Comment:
Consider renaming to reflect de-duplication here as well, e.g. `set_unique_list_attributes`.